client/gutenberg/extensions/slideshow/index.js

```diff
@@ -11,7 +11,7 @@ import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
  */
 import edit from './edit';
 import save from './save';
-import './style.scss';
+import './editor.scss';
```

Comment:
Shouldn't this instead be imported into `edit.js` (whereas we can keep the `style.scss` import here)? The elements that `editor.scss` is styling are all there, aren't they? (That'd also allow us to drop the [`style.scss` import](https://github.com/Automattic/wp-calypso/pull/30633/files#diff-e6b2e669b4ea9380f6359802c804390bR3) from `editor.scss`.)

Reply:
Importing the editor styles in `index.js` seems to be the approach taken in the other Jetpack blocks, e.g.: 

https://github.com/Automattic/wp-calypso/blob/master/client/gutenberg/extensions/markdown/index.js#L12
https://github.com/Automattic/wp-calypso/blob/master/client/gutenberg/extensions/tiled-gallery/index.js#L19

And importing the view styles into `editor.scss` also seems to be the pattern that's been used:

https://github.com/Automattic/wp-calypso/blob/master/client/gutenberg/extensions/tiled-gallery/editor.scss#L1

If you think this needs a new look, perhaps we should examine the pattern across all our blocks?

Reply:
Hmm. I love to keep concerns separate. I think our JS files might not map directly to other blocks? I.e. we have `slideshow.js` which is used both in view and edit, and thus needs the 'generic' slideshow styling (`style.scss`). In an editor context, where there are additional controls, it makes sense to import styling that's only dedicate to that separately.

I'd need to take a closer look at other blocks (e.g. TG) -- I think it'd be okay to deviate from what they're doing if their JS files really are differently organized.

The above would be my preferred way of organizing these files, but if you'd rather keep things as they're now, fine with me as well.

Reply:
I believe it's `create-swiper.js` that's used in view and edit, not `slideshow.js`—is that what you meant? Here's a commit in which `style.scss`  is imported only in `create-swiper.js` (which will be present in both environments). `editor.scss` is imported only in `edit.jsx` as you suggested. All other imports are gone, including the import of `style.scss` into `editor.scss`. So now there is only one import of each stylesheet, which should affect only the environments where they are needed. How's this look?

https://github.com/Automattic/wp-calypso/pull/30633/commits/887adefd6891c5ddfd89302a7a6deb4dabf2794a

Reply:
> I believe it's `create-swiper.js` that's used in view and edit, not `slideshow.js`—is that what you meant?

I did mean `slideshow.js` -- it's _where_ it's used that I made a mistake about: I meant edit and _save_, not _view_.

> Here's a commit in which `style.scss` is imported only in `create-swiper.js` (which will be present in both environments). `editor.scss` is imported only in `edit.jsx` as you suggested. All other imports are gone, including the import of `style.scss` into `editor.scss`. So now there is only one import of each stylesheet, which should affect only the environments where they are needed. How's this look?
> 
> [887adef](https://github.com/Automattic/wp-calypso/commit/887adefd6891c5ddfd89302a7a6deb4dabf2794a)

I :heart: it